mbed_tools_ci/generate_news.py

```diff
@@ -54,15 +54,20 @@ def _calculate_version(commit_type: CommitType, use_news_files: bool) -> \
         ConfigurationVariable.PROJECT_CONFIG)
     new_version: Optional[str] = None
     with cd(os.path.dirname(project_config_path)):
-        old, new_version, updates = auto_version_tool.main(
+        old, _, updates = auto_version_tool.main(
             release=is_release,
             enable_file_triggers=enable_file_triggers,
             bump=bump,
             config_path=project_config_path,
         )
+        # Autoversion second returned value is not actually the new version
+        # There seem to be a bug in autoversion.
+        # This is why the following needs to be done to determine the version
+        new_version = updates['__version__']
+    logger.info(f'{old}, {new_version}, {updates}')
     logger.info(':: Determining the new version')
     logger.info(f'Version: {new_version}')
-    return new_version
+    return old != new_version, new_version
```

Comment:
What's `old` and what's `new_version, new_version`? Can this ever be equal?

Reply:
D'oh. I get it now. Leaving first comment to illustrate how this can get confusing.
How about extracting `is_new_version` to a separate variable above?